llvm/lib/Transforms/IPO/OpenMPOpt.cpp

```diff
@@ -522,6 +522,7 @@ struct OMPInformationCache : public InformationCache {
   void recollectUses() {
     for (int Idx = 0; Idx < RFIs.size(); ++Idx)
       recollectUsesForFunction(static_cast<RuntimeFunction>(Idx));
+    OMPBuilder.Config.IsTargetDevice = isOpenMPDevice(OMPBuilder.M);
```

Comment:
Just curious, why do we need this? And why is this not set during the construction of the optimizer by the caller itself?

Reply:
OpenMP optimization passes uses OMPIRBuilder to generate parallel code. This patch enables generation of the parallel code for device and host. That's why we need to specify which version of the code we need to generate.

I moved this setting to the constructor of struct OMPInformationCache : public InformationCache.

Reply:
Okay, thanks for the explanation.